src/Providers/Api/Request/DataProviderAbstract.php

```diff
@@ -523,6 +526,27 @@ protected function getBankAccount()
         return $account->jsonSerialize();
     }
 
+    /**
+     * @param string $basketId
+     * @param $basketAmount
+     * @param string $currency
+     * @return array
+     */
+    protected function getAmazonPayData(string $basketId, $basketAmount, string $currency)
+    {
+        /** @var SessionStorage $sessionStorage */
+        $sessionStorage = pluginApp(SessionStorage::class);
+        $amazonAuthConfig = [];
+        $amazonAuthConfig['workOrderId'] = $sessionStorage->getSessionValue('workOrderId');
+        $amazonAuthConfig['amazonReferenceId'] = $sessionStorage->getSessionValue('amazonReferenceId');
+        $amazonAuthConfig['reference'] = $basketId;
+
+        $amazonAuthConfig['currency'] = $currency;
+        $amazonAuthConfig['amount'] = $basketAmount * 100;
```

Comment:
Warum mal 100?

Reply:
`Amount in smallest unit`. Hab es nochmal hinzugefügt als Kommentar.